### DISCUSSION TITLE
Why not have npm cache clear default to the --force behaviour?

### DISCUSSION THREAD

**Discussion by steveharrison** [npm/feedback#530] [CLI] [answered]
I find the default behaviour of `npm cache clear` perplexing. It's cool that NPM does a lot of nice stuff by default which means you shouldn't have to clear your cache as often, but when I ask to clear the cache... why not just clear the cache?

For context, clearing the cache is quite useful for me because I sometimes refer to modules on my local filesystem in my `package.json`, and every time I make a change to the referenced module, after re-building and re-packing it, I run an `npm cache clear --force` and then `npm i` in my main project to install the new version (my company's codebase has issues with `npm link`).

The only downsides of clearing the cache that I can think of are:
- You're trying to reproduce a bug and you want the cache to not change at all, which I would think is rare and in this scenario, you'd have the common-sense to not clear the cache.
- You have to download modules again. OK, re-downloading modules might take a few min extra, but I would hardly consider it a "dangerous" operation (when you use `npm cache clear --force`, it outputs a `I sure hope you know what you're doing` error message—why?).

So, in summary, why not just get rid of the `--force` option for `npm cache clear` and make this the default behaviour?

**Comment by MylesBorins** [comment, marked as answer]
@steveharrison if you would like to propose an explicit change to the npm CLI for this behavior the next step would be to open an rcs issue to propose the change to the CLI team

https://github.com/npm/rfcs
